src/main/js/src/components/Registration/RegistrationForm/RegistrationForm.js

```diff
@@ -318,10 +318,10 @@ export const registrationForm = props => {
       inputField(name, true, null, type)
     );
 
-  const showExamLang = () => {
-    const lang = props.initData.exam_session.language_code;
+  const showExamLangOptions = () => {
+    const { language_code, level_code } = props.initData.exam_session;
 
-    return !(lang === 'fin' || lang === 'swe');
+    return !['fin', 'swe', 'eng'].includes(language_code) || (language_code === 'eng' && level_code === 'PERUS');
```

Comment:
Tuntuu menevän aivot hieman solmuun tätä lukiessa. Eikös tää olisi paljon suoraviivaisempi jos logiikan kääntäisi toisin päin: funktion nimeksi esim `hideExamLangOptions` ja loppuun `return language_code === 'fin' || language_code === 'swe' || (language_code === 'eng' && level_code !== 'PERUS')`?

Reply:
Mulla itse asiassa oli tä näin aiemmin mut päädyin sit olemaan muuttamatta funktion nimeä ja laittamaan negaatiota `!` funktiokutsun yhteyeen. Mut joo, vois vaihtaa nyt näinkin.

Reply:
Tässä yhteydessä ei varmaan tehdä tuolle vielä mitään että SOLKI:in lähtee tieto suomen kielestä tehtävien kielenä vaikka oikeasti tehtävien kieli olisi englanti? Se logiikka tais olla myös tässä tiedostossa mukana?

Reply:
Joo, täällä kai pakotetaan tehtävien kieli suomeksi mikäli ei valittu kieleksi ruotsia. Pitäisi siis toimia tällaisenaan.